src/osd/PG.cc

```diff
@@ -3298,7 +3298,11 @@ void PG::reg_next_scrub()
       (info.stats.stats_invalid && g_conf->osd_scrub_invalid_stats)) {
     scrubber.scrub_reg_stamp = utime_t();
   } else {
-    scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    if (info.history.last_scrub_stamp + cct->_conf->osd_scrub_min_interval <
+        info.history.last_deep_scrub_stamp + cct->_conf->osd_deep_scrub_interval)
+      scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    else
+      scrubber.scrub_reg_stamp = info.history.last_deep_scrub_stamp;
```

Comment:
osd_deep_scrub_interval default is 7 days, osd_scrub_min_interval  is 1 day, So I think if last_deep_scrub_stamp  is much smaller than last_scrub_stamp, we should give the deep scrub pg much higher priority to be scheduled than normal scrub. So we could use last_deep_scrub_stamp instead of last_scrub_stamp.